f5lbaasdriver/v2/bigip/service_builder.py

```diff
@@ -201,6 +208,18 @@ def _get_extended_member(self, context, member):
 
         return (member_dict, subnet, network)
 
+    @log_helpers.log_method_call
+    def _get_extended_qos(self, context, loadbalancer):
+        try:
+            sess = keystone.get_session()
+            client_sess = client.Client(session=sess)
+            project = client_sess.projects.get(project=loadbalancer.tenant_id)
+            return project.qos
+
+        except Exception as e:
+            LOG.error('Exception: Get keystone project: %s', e.message)
+            return ''
```

Comment:
why do we need this `return ''`

Reply:
when the qos property doesn't exist, we return '' which means empty.

Reply:
if remove the `return`, the return value will still be empty, right?

Reply:
> if remove the `return`, the return value will still be empty, right?

I think so but it is better if we could explicitly return the ''?